compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ConstantPropagation.java

```diff
@@ -997,6 +999,23 @@ public void visit(BLangOrderByClause orderByClause) {
         result = orderByClause;
     }
 
+    @Override
+    public void visit(BLangGroupByClause groupByClause) {
+        groupByClause.groupingKeyList.forEach(value -> rewrite((BLangNode) value));
+        result = groupByClause;
+    }
+
+    @Override
+    public void visit(BLangGroupingKey groupingKey) {
+        BLangNode groupingKeyNode = (BLangNode) groupingKey.getGroupingKey();
+        if (groupingKeyNode.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+            groupingKey.variableRef = rewrite(groupingKey.variableRef);
+        } else {
+            groupingKey.variableDef = rewrite(groupingKey.variableDef);
+        }
```

Comment:
I think following is enough because we are doing null check in rewrite method.
```java
groupingKey.variableRef = rewrite(groupingKey.variableRef);
groupingKey.variableDef = rewrite(groupingKey.variableDef);
```

Reply:
Or else, instead of if-else block, we can do something like this,
`groupingKeyNode = rewrite(groupingKeyNode);`